pkg/bundle/parameters.go

```diff
@@ -3,23 +3,27 @@ package bundle
 import (
 	"errors"
 	"fmt"
+	"regexp"
 	"strconv"
 	"strings"
 )
 
 // ParameterDefinition defines a single parameter for a CNAB bundle
 type ParameterDefinition struct {
-	DataType      string             `json:"type" mapstructure:"type"`
-	DefaultValue  interface{}        `json:"defaultValue,omitempty" mapstructure:"defaultValue"`
-	AllowedValues []interface{}      `json:"allowedValues,omitempty" mapstructure:"allowedValues"`
-	Required      bool               `json:"required,omitempty" mapstructure:"required"`
-	MinValue      *int               `json:"minValue,omitempty" mapstructure:"minValue"`
-	MaxValue      *int               `json:"maxValue,omitempty" mapstructure:"maxValue"`
-	MinLength     *int               `json:"minLength,omitempty" mapstructure:"minLength"`
-	MaxLength     *int               `json:"maxLength,omitempty" mapstructure:"maxLength"`
-	Metadata      *ParameterMetadata `json:"metadata,omitempty" mapstructure:"metadata"`
-	Destination   *Location          `json:"destination,omitemtpty" mapstructure:"destination"`
-	ApplyTo       []string           `json:"apply-to,omitempty" mapstructure:"apply-to,omitempty"`
+	DataType         string            `json:"type" mapstructure:"type"`
+	DefaultValue     interface{}       `json:"defaultValue,omitempty" mapstructure:"defaultValue,omitempty"`
+	Required         bool              `json:"required" mapstructure:"required"`
+	Minimum          *int              `json:"minimum,omitempty" mapstructure:"minimum,omitempty"`
+	Maximum          *int              `json:"maximum,omitempty" mapstructure:"maximum,omitempty"`
+	ExclusiveMinimum *int              `json:"exclusiveMinimum,omitempty" mapstructure:"exclusiveMinimum,omitempty"`
+	ExclusiveMaximum *int              `json:"exclusiveMaximum,omitempty" mapstructure:"exclusiveMaximum,omitempty"`
+	MinLength        *int              `json:"minLength,omitempty" mapstructure:"minLength,omitempty"`
+	MaxLength        *int              `json:"maxLength,omitempty" mapstructure:"maxLength,omitempty"`
+	Pattern          string            `json:"pattern,omitempty" mapstructure:"pattern,omitempty"`
+	Enum             []interface{}     `json:"enum,omitempty" mapstructure:"enum,omitempty"`
+	Metadata         ParameterMetadata `json:"metadata,omitempty" mapstructure:"metadata,omitempty"`
```

Comment:
Should `Metadata` be `Description`?

Reply:
Looking at the [parameter definition from the spec](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#parameters):

- `metadata`: Holds fields that are not used in validation (OPTIONAL)
  - `description`: A user-friendly description of the parameter

And it's the same here.